### PR TITLE
ceph: add support to run all ceph command

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit add support to run all the ceph commands without the need of having toolbox pod but this will be hidden command and will not be visible in help section.

### CODE REVIEW

File: cmd/odf/ceph/ceph.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"github.com/spf13/cobra"
+)
+
+// CephCmd represents the ceph command
+var CephCmd = &cobra.Command{
+	Use:                "ceph",
+	Short:              "call a 'ceph' CLI command with arbitrary args",
+	DisableFlagParsing: true,
+	Hidden:             true,
```
Comment (subhamkrai) [inline cmd/odf/ceph/ceph.go:19]:
```suggestion
	Hidden:             true,
```
PTAL at this @travisn do we want this or not? Thanks

Comment (subhamkrai) [reply cmd/odf/ceph/ceph.go]: Based on the offline discussion removing this.

**Review by travisn — CHANGES_REQUESTED** [review]
A few additions to suggest:
- Add the new commands to the docs in this repo
- To make available as much functionality from the toolbox as possible, we also need to add the commands from the upstream cli: radosgw.go and rbd.go
- Test against an ODF cluster to confirm the commands work with the downstream operator

**Comment by subhamkrai** [comment]
> A few additions to suggest:
> 
> * Add the new commands to the docs in this repo
> * To make available as much functionality from the toolbox as possible, we also need to add the commands from the upstream cli: radosgw.go and rbd.go
> * Test against an ODF cluster to confirm the commands work with the downstream operator

Done

File: README.md [hunk 2]
```diff
@@ -4,35 +4,40 @@ The ODF CLI tool provides configuration and troubleshooting commands for OpenShi
...
-- `odf operator`:
-  - `rook`:
-    - `set`: Set the property in the rook-ceph-operator-config configmap.
-    - `restart` : Restart the Rook-Ceph operator
-- `odf restore`:
-  - `mon-quorum`: Restore the mon quorum based on a single healthy mon since quorum was lost with the other mons
-  - `deleted`: Restore the ceph resources which are stuck in deleting state due to underlying resources being present in the cluster
-- `odf help` : Display help text
+- `odf`:
+  - `ceph <args>` : Run a Ceph CLI command. Supports any arguments the ceph command supports. See [Ceph](https://docs.ceph.com/en/pacific/start/intro/) docs for more.
```
Comment (travisn) [inline README.md:8]: Let's add these new commands as the last commands in the list. They are advanced commands, so I'd rather the other commands be first in the doc. Also, to keep the diff simple how about not changing the structure of the other commands in this PR. The old structure looks ok to me, but feel free to open a separate PR for that if desired.

File: docs/ceph.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+# POOL_NAME     USED  OBJECTS  CLONES  COPIES  MISSING_ON_PRIMARY  UNFOUND  DEGRADED  RD_OPS      RD  WR_OPS       WR  USED COMPR  UNDER COMPR
+# .mgr       452 KiB        2       0       2                   0        0         0      22  18 KiB      14  262 KiB         0 B          0 B
+
+# total_objects    2
+# total_used       27 MiB
+# total_avail      10 GiB
+# total_space      10 GiB
+```
+
+This also supports all the rados supported flags like `--format json-pretty`
```
Comment (travisn) [inline docs/ceph.md:133]: I'd suggest let's keep this doc very simple, with just one example. So perhaps remove lines 133-164?

File: docs/ceph.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
+# Ceph
+
+This used to run any ceph cli command with with arbitrary args.
```
Comment (travisn) [inline docs/ceph.md:3]: How about an intro sentence to the overall doc?
```suggestion
Commands in this doc are an alternate to running the toolbox pod. 

## ceph

This used to run any ceph cli command with with arbitrary args.
```

File: docs/ceph.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
+# Ceph
```
Comment (travisn) [inline docs/ceph.md:1]:
```suggestion
# Ceph Commands
```

File: docs/ceph.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+#     osd: 1 osds: 1 up (since 12s), 1 in (since 30s)
+#
+#   data:
+#     pools:   0 pools, 0 pg
+#     objects: 0 objects, 0 B
+#     usage:   0 B used, 0 B / 0 B avail
+#     pgs:
+```
+
+This also supports all the ceph supported flags like `--format json-pretty`
```
Comment (travisn) [inline docs/ceph.md:26]: Remove the second example, just one example needed for each command IMO

File: docs/ceph.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+#     "servicemap": {
+#         "epoch": 1,
+#         "modified": "2022-09-14T08:55:39.603658+0000",
+#         "services": {}
+#     },
+#     "progress_events": {}
+# }
+```
+
+## Daemon
```
Comment (travisn) [inline docs/ceph.md:100]: The title can be the command name, including lower case
```suggestion
## ceph daemon
```

File: docs/ceph.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+Info: running 'ceph' command with args: [daemon osd.0 dump_historic_ops]
+{
+    "size": 20,
+    "duration": 600,
+    "ops": []
+}
+```
+
+# Rados
```
Comment (travisn) [inline docs/ceph.md:115]:
```suggestion
# rados
```

File: docs/ceph.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+This used to run any rados cli command with with arbitrary args.
+
+## Examples
```
Comment (travisn) [inline docs/ceph.md:119]:
```suggestion
## Example
```

File: docs/ceph.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+#         }
+#     ],
+#     "total_objects": 2,
+#     "total_used": 27404,
+#     "total_avail": 10458356,
+#     "total_space": 10485760
+# }
+```
+
+# RBD
```
Comment (travisn) [inline docs/ceph.md:166]:
```suggestion
# rbd
```

**Review by travisn — CHANGES_REQUESTED** [review]
Some doc suggestions...

File: docs/ceph.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
+Commands in this doc are an alternate to running the toolbox pod.
+
+# Ceph Commnads
```
Comment (travisn) [inline docs/ceph.md:3]: I was thinking to have the level 1 header at the top, then each level-2 header would have the name of the command, like this:
```
# Ceph Commnads

Commands in this doc are an alternate to running the toolbox pod.

## ceph
```

**Review by travisn — CHANGES_REQUESTED** [review]
LGTM, just one more nit on the doc. Could you also confirm it was successful testing against an openshift cluster?

**Comment by subhamkrai** [comment]
> LGTM, just one more nit on the doc. Could you also confirm it was successful testing against an openshift cluster?

Yes I have tested on openshift cluster

File: docs/ceph.md (hunk 11 above)
Comment (subhamkrai) [reply docs/ceph.md]: DOne

File: docs/ceph.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+Commands in this doc are an alternate to running the toolbox pod.
+
+## ceph
+
+This used to run any ceph cli command with with arbitrary args.
```
Comment (travisn) [inline docs/ceph.md:7]:
```suggestion
Run the `ceph` cli command with with arbitrary args.
```

File: docs/ceph.md [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+{
+    "size": 20,
+    "duration": 600,
+    "ops": []
+}
+```
+
+## rados
+
+This used to run any rados cli command with with arbitrary args.
```
Comment (travisn) [inline docs/ceph.md:47]:
```suggestion
Run any `rados` cli command with with arbitrary args.
```

File: docs/ceph.md [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+
+# total_objects    2
+# total_used       27 MiB
+# total_avail      10 GiB
+# total_space      10 GiB
+```
+
+## rbd
+
+This used to run any rbd cli command with with arbitrary args.
```
Comment (travisn) [inline docs/ceph.md:65]:
```suggestion
Run any `rbd` cli command with with arbitrary args.
```

File: docs/ceph.md [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+
+```bash
+odf rbd ls replicapool
+
+# csi-vol-427774b4-340b-11ed-8d66-0242ac110004
+```
+
+## radosgw-admin
+
+This runs any radosgw-admin CLI command with arbitrary args.
```
Comment (travisn) [inline docs/ceph.md:77]:
```suggestion
Run any `radosgw-admin` CLI command with arbitrary args.
```

**Review by travisn — COMMENTED** [review]
A few final nits

File: docs/ceph.md (hunk 15 above)
Comment (subhamkrai) [reply docs/ceph.md]: done

**Review by travisn — APPROVED** [review]
/lgtm
/approve

**Pull request merged**